Framework/DataHandling/src/LoadInstrument.cpp

```diff
@@ -150,8 +152,21 @@ void LoadInstrument::exec() {
     const std::string::size_type stripPath = filename.find_last_of("\\/");
     std::string instrumentFile =
         filename.substr(stripPath + 1, filename.size());
+
     // Strip off "_Definition.xml"
-    instname = instrumentFile.substr(0, instrumentFile.find("_Def"));
+    auto definitionRange = boost::ifind_first(instrumentFile, "_Def");
+    if (definitionRange) {
+      // We need to find the position this occurs in the string and we
```

Comment:
Can you use `std::distance` with something like:
```c++
instname = filename.substr(0, std::distance(filename.begin(), range.begin()));
```
?

Reply:
I knew the STL must have a way to handle the difference between a normal and string iterator, but just couldn't find it. Thanks for that.